src/rpc/mining.cpp

```diff
@@ -702,7 +702,7 @@ static RPCHelpMan getblocktemplate()
             std::string lpstr = lpval.get_str();
 
             hashWatchedChain = ParseHashV(lpstr.substr(0, 64), "longpollid");
-            nTransactionsUpdatedLastLP = atoi64(lpstr.substr(64));
+            nTransactionsUpdatedLastLP = LocaleIndependentAtoi<int64_t>(lpstr.substr(64));
```

Comment:
`nTransactionsUpdatedLastLP` is `unsigned int`; should this be defined or cast to that?

Reply:
Yes it probably needs the same kind of ugly cast as `nTimeSmart` in the wallet code. In the longer run it should imo be converted to an explicitly sized type, and parsed as that, but keeping the behavior exactly the same makes sense here to keep it uncontroversial.

Reply:
@jonatack Good point! I suggest we tackle that in a follow-up PR to keep this PR refactoring only :)